### PR TITLE
Bug fix - COND_InitialOnly not doing anything

### PR DESCRIPTION
**Pull request by BenNaccarato**
#### Description
The bug was that there was some native Unreal code that would check an OpenPacketId, and use this to mark the repflag as being the initial update. However, since OpenPacketId is not used in spatial, this is simply always true, so initial only updates are sent every time. This PR overrides the native check with spatial's CreatingNewEntity.

To note - there is some legacy native code that could be removed (regarding a FOutBunch). This could be relevant to look into with this bug fix, but maybe it's better to ignore for now.

#### Release note
Bugfix - COND_InitialOnly are only replicated once at the start.

#### Tests
Fixes the bug reported in UT

#### Primary reviewers
@m-samiec @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -260,6 +260,9 @@ int64 USpatialActorChannel::ReplicateActor()
 		Bunch.bReliable = true; // Net temporary sends need to be reliable as well to force them to retry
 	}
 
+	// Use spatial's bCreatingNewEntity to inform the replication flags.
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: Slight driveby, but could we remove the `OpenPacketId` stuff above?

Comment (BenNaccarato) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: So a slight question with this would be that there is some code within that logic regarding a FOutBunch. Since the OpenPacketId case is always reached, it would be effectively the same to take the code out of the if statement. It may be the case that we don't need the Bunch at all however, but it is passed to a virtual OnSerializeNewActor method, and I'm currently unclear on how this would be intended to be used, and whether removing this entirely would lead to unexpected consequences down the line

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: All bunch usage only remains to satisfy existing interfaces. And we're now overriding the RepFlags.bNetInitial, so removing the OpenPacketId clause is safe to remove.
